app/src/main/java/de/othr/plantico/ui/plantDetails/PlantDetailActivity.kt

```diff
@@ -0,0 +1,151 @@
+        layout.flexDirection = FlexDirection.ROW
+        layout.justifyContent = JustifyContent.FLEX_START
+        binding.categoriesItemList.layoutManager = layout
+        val adapter = CategorieListAdapter(
+            this
+        )
+        binding.categoriesItemList.adapter = adapter
+
+        var plant: Plant? = null
+        plantViewModel.allPlants.observe(this) { plants ->
+            // Update the cached copy of the words in the adapter.
+            plants.let {
+                plantList.clear()
+                plantList.addAll(it)
+                plant = plantList.find { plant -> plant.id == plantID }!!
+                if (plant != null) {
+
+                    binding.apply {
+                        plantName.text = plant!!.plantName
+                        plantFamilyText.text = plant!!.family
+                        plantSubfamilyText.text = plant!!.subfamily
+                        plantTribeText.text = plant!!.plantNameLat
+                        plantGenusText.text = plant!!.genus
+                        plantSpeciesText.text = plant!!.typeSpecies
+                        adapter.setList(plant!!.plantCategory)
+                        plantWateringCycleText.text = plant!!.wateringCycleDays.toString() + " Days"
+                        plantGeneralDescriptionText.text = plant!!.desc
+                        plantPropogationsInstructionsText.text = plant!!.propagationDesc
+                        plantWateringInstructionsText.text = plant!!.wateringDesc
+                        //Difficulty
+                        //1 equals PlantDifficulty.INTERMEDIATE
+                        if (plant!!.difficulty.ordinal >= 1) {
+                            plantDifficultyMedium.visibility = View.VISIBLE
+                        }
+                        //2 equals PlantDifficulty.ADVANCED
+                        if (plant!!.difficulty.ordinal >= 2) {
+                            plantDifficultyHard.visibility = View.VISIBLE
+                        }
+                        //Watering
+                        if (plant!!.wateringLevel.ordinal < 2) {
+                            wateringHeavy.setImageDrawable(
+                                ContextCompat.getDrawable(
+                                    application,
+                                    R.drawable.ic_waterdropplett_empty_icon
+                                )
+                            )
+                        }
+                        if (plant!!.wateringLevel.ordinal < 1) {
+                            wateringMedium.setImageDrawable(
+                                ContextCompat.getDrawable(
+                                    application,
+                                    R.drawable.ic_waterdropplett_empty_icon
+                                )
+                            )
+                        }
+                    }
+                }
+            }
+        }
+        setupMenuBinding()
+    }
+
+    fun setupMenuBinding() {
```

Comment:
Hmmm, maybe we should consider to outsource this since we have the same piece of code in most activities here.

Reply:
Good idea. Did that.